siad/daemon.go

```diff
@@ -295,6 +377,9 @@ func startDaemon(config Config) (err error) {
 	// connect the API to the server
 	srv.mux.Handle("/", a)
 
+	// Attempt to auto-unlock the wallet
+	walletAutoUnlock(config)
```

Comment:
It might make more sense to just pass `w` to `walletAutoUnlock` and use `(*wallet).Unlock` directly instead of duplicating all of this API call code from siac. Check out `walletUnlockHandler` in `api/wallet.go` for an example of how to extract a `crypto.TwofishKey` from a string.

Reply:
@avahowell that actually makes alot more sense than the way I'm doing it. I'll get on that PR later tonight.

Reply:
I think using the API handler is better actually, since it more closely replicates the behavior of siac. But instead of duplicating the HTTP code, you can use `api.HttpPOST` and `api.HttpPOSTAuthenticated`.

Reply:
I guess it's kind of a wash. Calling the HTTP endpoint means you don't have to do the awkward multiple-password handling. But it also means you're calling an HTTP endpoint, and it's undeniably a bit weird for a server to call its own endpoints.

Reply:
@lukechampine with the HTTP endpoint, we pass english as the query param, but technically there could be another dictionary right? 

I don't think this commit covers the other languages, since english is hard coded as the param passed to dictionary.